core/iwasm/compilation/aot_llvm.c

```diff
@@ -1004,17 +1004,24 @@ create_aux_stack_info(const AOTCompContext *comp_ctx, AOTFuncContext *func_ctx)
 
     if (!(aux_stack_bound_addr =
               LLVMBuildBitCast(comp_ctx->builder, aux_stack_bound_addr,
-                               INT32_PTR_TYPE, "aux_stack_bound_ptr"))) {
+                               INTPTR_T_PTR_TYPE, "aux_stack_bound_ptr"))) {
         aot_set_last_error("llvm build bit cast failed");
         return false;
     }
 
     if (!(func_ctx->aux_stack_bound =
-              LLVMBuildLoad2(comp_ctx->builder, I32_TYPE, aux_stack_bound_addr,
-                             "aux_stack_bound"))) {
+              LLVMBuildLoad2(comp_ctx->builder, INTPTR_T_TYPE,
+                             aux_stack_bound_addr, "aux_stack_bound_intptr"))) {
         aot_set_last_error("llvm build load failed");
         return false;
     }
+    /* TODO: Memory64 the actual type depends on mem_idx type */
+    if (!(func_ctx->aux_stack_bound = LLVMBuildTruncOrBitCast(
+              comp_ctx->builder, func_ctx->aux_stack_bound, I32_TYPE,
+              "aux_stack_bound_mem_idx_type"))) {
```

Comment:
I found the aux_stack_bound is extended to I64_TYPE to check in wasm_interp_classic.c, is it better to change to I64_TYPE here and change the check in the translation of WASM_OP_SET_GLOBAL_AUX_STACK (in aot_emit_variable.c)? So it seems we don't need to do anything for when memory64 is enabled as the aux global is 64-bit in memory64.